File: core/workflow-operator/src/main/scala/edu/uci/ics/amber/operator/filter/SpecializedFilterOpExec.java
package edu.uci.ics.amber.operator.filter;

import edu.uci.ics.amber.core.tuple.Tuple;
import scala.Function1;

import java.io.Serializable;

public class SpecializedFilterOpExec extends FilterOpExec {

    private final java.util.List<edu.uci.ics.amber.operator.filter.FilterPredicate> predicates;

    public SpecializedFilterOpExec(java.util.List<edu.uci.ics.amber.operator.filter.FilterPredicate> predicates) {
        this.predicates = predicates;
        setFilterFunc((Function1<Tuple, Boolean> & Serializable) this::filterFunc);
    }

    public Boolean filterFunc(Tuple tuple) {
        return predicates.stream().anyMatch(predicate -> predicate.evaluate(tuple));
    }

}

File: core/workflow-operator/src/main/scala/edu/uci/ics/amber/operator/filter/SpecializedFilterOpExec.scala
package edu.uci.ics.amber.operator.filter

import edu.uci.ics.amber.core.tuple.Tuple
import edu.uci.ics.amber.util.JSONUtils.objectMapper

class SpecializedFilterOpExec(descString: String) extends FilterOpExec {
  private val desc: SpecializedFilterOpDesc =
    objectMapper.readValue(descString, classOf[SpecializedFilterOpDesc])
  setFilterFunc((tuple: Tuple) => desc.predicates.exists(_.evaluate(tuple)))
}
